shim/src/realm.js

```diff
@@ -1,61 +1,226 @@
-import { createSandbox } from "./sandbox.js";
-import { sanitize } from "./sanitize.js";
+import { createSandbox, setSandboxGlobalObject } from "./sandbox.js";
 import { evaluate } from "./evaluate.js";
-import { getEvaluators } from "./evaluators.js";
 import { getStdLib } from "./stdlib.js";
 import { getIntrinsics } from "./intrinsics.js";
-import { proxyHandler } from "./proxy.js";
+import { assert, IsCallable } from "./utils";
 
-const RealmToSandbox = new WeakMap();
+const RealmRecord = Symbol('Realm Slot');
+const Intrinsics = Symbol('Intrinsics Slot');
+const GlobalObject = Symbol('GlobalObject Slot');
+const GlobalThisValue = Symbol('GlobalThisValue Slot');
+const GlobalEnv = Symbol('GlobalEnv Slot');
+const EvalHook = Symbol('EvalHook Slot');
+const IsDirectEvalHook = Symbol('IsDirectEvalHook Slot');
+const ImportHook = Symbol('ImportHook Slot');
+const ImportMetaHook = Symbol('ImportMetaHook Slot');
+const ShimSandbox = Symbol('Sandbox');
 
-function getSandbox(realm) {
-    const sandbox = RealmToSandbox.get(realm);
-    if (!sandbox) {
-        throw new Error(`Invalid realm object.`);
-    }
+// shim specific
+function getSandbox(realmRec) {
+    const sandbox = realmRec[ShimSandbox];
+    assert(typeof sandbox === 'object');
     return sandbox;
 }
 
+function getCurrentRealmRecord() {
```

Comment:
I'm not really using this just yet, but it is suppose to help with the inheritance of intrinsics